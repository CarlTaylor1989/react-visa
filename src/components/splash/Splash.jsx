import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-redux-i18n';
import { GoToScreen, newScreenConnect } from '../generic/GoToScreen';
import { splashPath } from '../../config/navigation';
import { splashReferrer } from '../../config/referrers';
import AnimatedText from '../generic/animatedtext/AnimatedText';
import splashJson from '../../translations/en_us/splash.json';

import './Splash.scss';

export class Splash extends GoToScreen {
  constructor(props) {
    super(props);

    this.state = {
      isIntroAnimationComplete: false,
      playIntroAnimation: false
    };
  }

  componentDidMount() {
    const { setPopupReferrer } = this.props;
    setPopupReferrer(splashReferrer);
    this.motivationalTexts = splashJson.motivationalTexts;
    this.setState({
      playIntroAnimation: true
    });
  }

  componentWillUnmount() {
    const { setScreenReferrer } = this.props;
    setScreenReferrer(splashReferrer);
    this.setState({
      playIntroAnimation: false
    });
  }

  render() {
    const { diagnosticCompleted, diagnosticVisited, lessonLocation } = this.props;
    const { isIntroAnimationComplete, playIntroAnimation } = this.state;

    return (
      <div className="splash">
        <div className="logo-container">
          <span className="small-logo" />
        </div>
        <div className="splash-content">
          {
            this.motivationalTexts
            && (
              <AnimatedText
                texts={this.motivationalTexts}
                playAnimation={playIntroAnimation}
                handleAnimationEnded={() => this.setState({ isIntroAnimationComplete: true })}
              />
            )
          }
          {(['', '/', splashPath].includes(lessonLocation)
            && !diagnosticCompleted && !diagnosticVisited ? (
              <button
                type="button"
                className={`nextBtn${isIntroAnimationComplete ? '' : ' hide'}`}
                onClick={this.goToNextScreen}
                disabled={!isIntroAnimationComplete}
              >
                <Translate value="splash.startBtn" />
              </button>
            ) : (
              <button
                type="button"
                className={`nextBtn${isIntroAnimationComplete ? '' : ' hide'}`}
                onClick={this.findAndGoToScreen}
                disabled={!isIntroAnimationComplete}
              >
                <Translate value="splash.continueBtn" />
              </button>
            ))}
        </div>
      </div>
    );
  }
}

Splash.propTypes = {
  lessonLocation: PropTypes.string.isRequired,
  setPopupReferrer: PropTypes.func.isRequired,
  setScreenReferrer: PropTypes.func.isRequired,
  suspendData: PropTypes.object.isRequired
};

export const mapStateToProps = state => ({
  suspendData: state.scormData.suspendData
});

export default newScreenConnect(Splash, mapStateToProps);
