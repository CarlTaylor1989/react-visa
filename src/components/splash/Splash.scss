@import '../../styles/buttons';
@import '../../styles/generic';

$vu-logo-icon: '../../styles/icons/splash/vu_logo.svg';
$vu-logo-icon-height: 50px;
$vu-logo-icon-width: 290px;

.splash {
  @extend %screen-size;
  @extend %bg-size;
  background-image: url('../../styles/backgrounds/splash_bg.jpg');
  position: relative;
  text-align: center;

  .logo-container {
    display: flex;
    flex-direction: column;
    padding: 65px 0 0 80px;

    .small-logo {
      align-self: start;
      background-image: url($vu-logo-icon);
      height: $vu-logo-icon-height;
      width: $vu-logo-icon-width;
    }
  }

  .nextBtn {
    @extend %blue-btn;
    @include opacity(1);
    align-self: center;
    border: 3px solid $colour-white;
    font-size: 46px;
    line-height: 50px;
    margin: 0 auto;
    min-width: 240px;
    padding: 15px 40px;
    transition: background .3s ease-in-out, opacity .3s ease-in-out;

    &.hide {
      @include opacity(0);
      visibility: hidden;
    }
  }

  .splash-content {
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: center;
  }
}

