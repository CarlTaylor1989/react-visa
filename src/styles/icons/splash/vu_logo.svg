<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="290" height="50" viewBox="0 0 290 50">
  <defs>
    <clipPath id="clip-vu_logo">
      <rect width="290" height="50"/>
    </clipPath>
  </defs>
  <g id="vu_logo" clip-path="url(#clip-vu_logo)">
    <g id="Group_5237" data-name="Group 5237" transform="translate(-404.615 -558.446)">
      <path id="Path_9619" data-name="Path 9619" d="M417.092,596l-12.477-37.555H407.4l6.629,20.115c1.837,5.46,3.341,10.2,4.4,14.377h.166c1.114-4.234,2.84-9.083,4.79-14.377l7.3-20.115h2.84L419.6,596Z" transform="translate(0 0)" fill="#fff"/>
      <path id="Path_9620" data-name="Path 9620" d="M424.371,561.2a2.177,2.177,0,0,1-2.227,2.283,2.147,2.147,0,0,1-2.062-2.283,2.287,2.287,0,0,1,2.172-2.341,2.173,2.173,0,0,1,2.117,2.341m-3.453,7.634h2.618v26.744h-2.618Z" transform="translate(15.927 0.424)" fill="#fff"/>
      <path id="Path_9621" data-name="Path 9621" d="M425.021,587.319a11.83,11.83,0,0,0,6.014,1.837c4.234,0,6.239-2.284,6.239-5.068,0-2.953-1.782-4.514-5.626-6.073-4.4-1.673-6.794-4.068-6.794-7.3,0-3.844,3.008-7.244,8.243-7.244a10.593,10.593,0,0,1,5.9,1.675l-1.057,2.172a8.661,8.661,0,0,0-5.292-1.616c-3.4,0-5.182,2.172-5.182,4.624,0,2.787,1.894,4.013,5.515,5.517,4.346,1.67,6.909,3.844,6.909,7.912,0,4.569-3.566,7.691-9.026,7.691a13,13,0,0,1-6.8-1.9Z" transform="translate(20.036 5.173)" fill="#fff"/>
      <path id="Path_9622" data-name="Path 9622" d="M452.585,584.477a47.731,47.731,0,0,0,.39,6.351H450.58l-.392-3.844h-.166a10.022,10.022,0,0,1-8.521,4.457c-5.348,0-7.855-3.79-7.855-7.354,0-6.185,5.458-9.918,16.321-9.806v-.67c0-2.675-.5-7.969-6.852-7.912a11.38,11.38,0,0,0-6.684,2.006l-.834-1.951a14.482,14.482,0,0,1,7.8-2.286c7.741,0,9.191,5.8,9.191,10.589Zm-2.618-7.969c-5.85-.167-13.536.726-13.536,7.133,0,3.846,2.564,5.572,5.346,5.572a8.049,8.049,0,0,0,7.912-5.348,5.81,5.81,0,0,0,.278-1.616Z" transform="translate(29.894 5.173)" fill="#fff"/>
      <path id="Path_9623" data-name="Path 9623" d="M451.525,558.446v22.679c0,9.528,4.457,13.206,10.025,13.206,6.185,0,10.473-4.013,10.473-13.206V558.446H474.7v22.4c0,11.533-6.237,15.769-13.311,15.769-6.574,0-12.479-3.846-12.479-15.323V558.446Z" transform="translate(45.608 0)" fill="#f7b600"/>
      <path id="Path_9624" data-name="Path 9624" d="M465.146,570.492c0-2.452-.112-4.289-.221-6.408h2.45l.223,4.847h.112a10.032,10.032,0,0,1,9.189-5.462c2.563,0,9.3,1.283,9.3,11.312v16.047h-2.618V575c0-4.847-1.9-9.25-7.409-9.25a8.542,8.542,0,0,0-8.021,6.13,7.982,7.982,0,0,0-.39,2.4v16.547h-2.618Z" transform="translate(62.102 5.173)" fill="#f7b600"/>
      <path id="Path_9625" data-name="Path 9625" d="M482.677,561.2a2.177,2.177,0,0,1-2.227,2.283,2.145,2.145,0,0,1-2.06-2.283,2.285,2.285,0,0,1,2.17-2.341,2.173,2.173,0,0,1,2.117,2.341m-3.453,7.634h2.618v26.744h-2.618Z" transform="translate(75.967 0.424)" fill="#f7b600"/>
      <path id="Path_9626" data-name="Path 9626" d="M484.841,563.772l6.237,16.606c1.058,2.618,1.894,4.9,2.618,7.131h.11c.781-2.229,1.729-4.569,2.73-7.131l6.349-16.606h2.787l-10.861,26.744h-2.34L482,563.772Z" transform="translate(79.684 5.485)" fill="#f7b600"/>
      <path id="Path_9627" data-name="Path 9627" d="M496.737,577.178c0,8.58,4.678,11.923,10.191,11.923a15.56,15.56,0,0,0,7.408-1.5l.67,2.115a17.631,17.631,0,0,1-8.411,1.727c-7.687,0-12.477-5.627-12.477-13.484,0-8.915,5.123-14.489,12.03-14.489,8.746,0,10.416,8.247,10.416,12.093a6.62,6.62,0,0,1-.112,1.616Zm17.043-2.117c.055-3.733-1.5-9.418-7.967-9.418-5.9,0-8.466,5.294-8.911,9.418Z" transform="translate(92.164 5.173)" fill="#f7b600"/>
      <path id="Path_9628" data-name="Path 9628" d="M507.762,572.107c0-2.675-.11-5.462-.223-8.024h2.452l.112,5.405h.112c1.169-3.343,3.954-6.02,7.63-6.02a6.987,6.987,0,0,1,1.059.112v2.566a4.563,4.563,0,0,0-1.281-.112c-3.678,0-6.294,3.177-7.019,7.466a18.271,18.271,0,0,0-.223,2.564v14.764h-2.618Z" transform="translate(105.983 5.173)" fill="#f7b600"/>
      <path id="Path_9629" data-name="Path 9629" d="M515.048,587.319a11.831,11.831,0,0,0,6.016,1.837c4.234,0,6.237-2.284,6.237-5.068,0-2.953-1.782-4.514-5.624-6.073-4.4-1.673-6.8-4.068-6.8-7.3,0-3.844,3.006-7.244,8.243-7.244a10.61,10.61,0,0,1,5.906,1.675l-1.06,2.172a8.656,8.656,0,0,0-5.292-1.616c-3.4,0-5.18,2.172-5.18,4.624,0,2.787,1.894,4.013,5.515,5.517,4.344,1.67,6.905,3.844,6.905,7.912,0,4.569-3.564,7.691-9.024,7.691a12.987,12.987,0,0,1-6.793-1.9Z" transform="translate(112.741 5.173)" fill="#f7b600"/>
      <path id="Path_9630" data-name="Path 9630" d="M528.186,561.2a2.179,2.179,0,0,1-2.229,2.283,2.146,2.146,0,0,1-2.06-2.283,2.286,2.286,0,0,1,2.172-2.341,2.174,2.174,0,0,1,2.117,2.341m-3.452,7.634h2.616v26.744h-2.616Z" transform="translate(122.827 0.424)" fill="#f7b600"/>
      <path id="Path_9631" data-name="Path 9631" d="M534.731,560.862v5.907h7.634v2.117h-7.634v17.385c0,3.4,1.06,5.572,3.9,5.572a9.682,9.682,0,0,0,3.063-.447l.333,2.062a9.311,9.311,0,0,1-3.733.668,5.58,5.58,0,0,1-4.567-1.949c-1.224-1.394-1.671-3.623-1.671-6.353V568.886h-4.51v-2.117h4.51v-4.9Z" transform="translate(126.586 2.488)" fill="#f7b600"/>
      <path id="Path_9632" data-name="Path 9632" d="M538.834,563.772l7.019,17.495c.725,1.9,1.449,4.013,1.949,5.74h.112c.558-1.672,1.226-3.733,2.062-5.9l6.515-17.33h2.84l-8.02,19.947c-3.231,8.3-5.515,13.263-9.022,16.325a15.248,15.248,0,0,1-4.681,2.787l-.834-2.229a14.027,14.027,0,0,0,4.678-2.953,21.285,21.285,0,0,0,4.455-6.741,3.835,3.835,0,0,0,.335-1.171,3.45,3.45,0,0,0-.335-1.169l-9.913-24.8Z" transform="translate(135.284 5.485)" fill="#f7b600"/>
    </g>
  </g>
</svg>
