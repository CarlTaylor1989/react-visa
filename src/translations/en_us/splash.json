{
  "startBtn": "START",
  "continueBtn": "CONTINUE",
  "motivationalTexts": [{
    "id": "one",
    "text": "The way we sell to our clients<br />is changing…"
    }, {
      "id": "two",
      "text": "We must provide solutions,<br />not just products…"
    }, {
      "id": "three",
      "text": "How you sell will shape the<br />future of our business…"
    }, {
      "id": "four",
      "text": "Are you ready for<br />the challenge?"
    }
  ]
}
