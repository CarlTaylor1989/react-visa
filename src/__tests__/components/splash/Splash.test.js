/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import { I18n } from 'react-redux-i18n';
import history from '../../../__mocks__/routerHistoryMock';
import { Splash, mapStateToProps } from '../../../components/splash/Splash';

I18n.t = jest.fn(() => ({
  0: {
    id: 'one',
    text: 'MESSAGE ONE MOTIVATIONAL MESSAGE'
  },
  1: {
    id: 'two',
    text: 'MESSAGE TWO MOTIVATIONAL MESSAGE'
  },
  2: {
    id: 'three',
    text: 'MESSAGE THREE MOTIVATIONAL MESSAGE'
  },
  3: {
    id: 'four',
    text: 'MESSAGE FOUR MOTIVATIONAL MESSAGE'
  }
}));

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    history,
    diagnosticCompleted: false,
    diagnosticVisited: false,
    lessonLocation: '',
    setLessonLocation: jest.fn(),
    setPopupReferrer: jest.fn(),
    setScreenReferrer: jest.fn(),
    suspendData: {}
  };
});

test('renders the Splash component with a start button', () => {
  const wrapper = shallow(<Splash {...mockProps} />);
  wrapper.instance().goToNextScreen = jest.fn();
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method componentWillUnmount', () => {
  const wrapper = shallow(<Splash {...mockProps} />, {
    disableLifecycleMethods: true
  });
  wrapper.instance().componentWillUnmount();
  expect(mockProps.setScreenReferrer).toHaveBeenCalledWith(expect.any(String));
});

test('renders the Splash component with a continue button', () => {
  mockProps.lessonLocation = '/diagnostic';
  mockProps.diagnosticVisited = true;
  const wrapper = shallow(<Splash {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('next button should have class "hide" until animation is complete', () => {
  const wrapper = shallow(<Splash {...mockProps} />);

  expect(wrapper.find('.nextBtn').hasClass('hide')).toBe(true);

  wrapper.instance().setState({ isIntroAnimationComplete: true, playIntroAnimation: true });

  expect(wrapper.find('.nextBtn').hasClass('hide')).toBe(false);
});

test('should mapStateToProps', () => {
  const state = {
    scormData: {
      suspendData: { foo: 'bar' }
    }
  };
  const props = mapStateToProps(state);
  expect(props).toEqual({
    ...state.scormData
  });
});
