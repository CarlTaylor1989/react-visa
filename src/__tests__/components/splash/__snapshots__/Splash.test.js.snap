// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Splash component with a continue button 1`] = `
<div
  className="splash"
>
  <div
    className="logo-container"
  >
    <span
      className="small-logo"
    />
  </div>
  <div
    className="splash-content"
  >
    <AnimatedText
      handleAnimationEnded={[Function]}
      playAnimation={true}
      texts={
        Array [
          Object {
            "id": "one",
            "text": "The way we sell to our clients<br />is changing…",
          },
          Object {
            "id": "two",
            "text": "We must provide solutions,<br />not just products…",
          },
          Object {
            "id": "three",
            "text": "How you sell will shape the<br />future of our business…",
          },
          Object {
            "id": "four",
            "text": "Are you ready for<br />the challenge?",
          },
        ]
      }
    />
    <button
      className="nextBtn hide"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <Translate
        tag="span"
        value="splash.continueBtn"
      />
    </button>
  </div>
</div>
`;

exports[`renders the Splash component with a start button 1`] = `
<div
  className="splash"
>
  <div
    className="logo-container"
  >
    <span
      className="small-logo"
    />
  </div>
  <div
    className="splash-content"
  >
    <AnimatedText
      handleAnimationEnded={[Function]}
      playAnimation={true}
      texts={
        Array [
          Object {
            "id": "one",
            "text": "The way we sell to our clients<br />is changing…",
          },
          Object {
            "id": "two",
            "text": "We must provide solutions,<br />not just products…",
          },
          Object {
            "id": "three",
            "text": "How you sell will shape the<br />future of our business…",
          },
          Object {
            "id": "four",
            "text": "Are you ready for<br />the challenge?",
          },
        ]
      }
    />
    <button
      className="nextBtn hide"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <Translate
        tag="span"
        value="splash.startBtn"
      />
    </button>
  </div>
</div>
`;
